### PR TITLE
Jdaviz Launcher: Identify compatible configs and request user to select config

### PR DESCRIPTION
**Pull request by duytnguyendtn**
- [x] Fix change log entry introduced in https://github.com/spacetelescope/jdaviz/pull/2257 . Need to mention CLI `--layout` that is reintroduced.

### Description

This pull request extends #2257 when a filepath is provided, identify the file and enable/disable the relevant config buttons that are compatible. Then allow the user to select the configuration they would like to load with the data selected:

https://github.com/spacetelescope/jdaviz/assets/25206008/fd4855de-e866-437f-a2f8-d6edc8fe05a5

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
Some quality-of-life improvement ideas:

* There needs to be some user feedback when a file is not found. Right now, it does nothing and also does not emit any error message. For example, what if I accidentally think the instance is launched from my working dir like normal notebook (it is not, it is in a temp dir but this is hidden from user) and I entered the filename without path?
* There needs to be some basic instructions on how to use the UI elements in the launcher itself. Having to enter text field by hand and then click on 🔍 is not a common workflow (for example, you don't do this in Photoshop).
* Make it clearer in the launcher that if you click on the viz button with blank text field, you are going to launch empty viz and lose the launcher.

File: CHANGES.rst [hunk 1]
```diff
@@ -13,7 +13,9 @@ New Features
 
 - The ``specviz.load_spectrum`` method is deprecated; use ``specviz.load_data`` instead. [#2273]
 
-- Add first-pass launcher to select config and auto-identify data. [#2257]
+- Add launcher to select config and require --layout argument when launching standalone. [#2257]
+
+- Launcher displays which configs are compatible with a provided dataset. [#2267]
```
Comment (pllim) [inline CHANGES.rst:18]: I think these two entries can be merged.

Comment (duytnguyendtn) [reply CHANGES.rst]: Good idea, done in https://github.com/spacetelescope/jdaviz/pull/2267/commits/942751809f54a6c8c7abb712f9ea295158bd8c39

**Review by kecnry — APPROVED** [review]
Seems reasonable to me.  Discussing offline, we'll defer the decision to migrate this logic to a full-fledged vue-template to the UI work, at which point we can iterate further on UX details.

**Comment by pllim** [comment]
> Again, all great points, I just feel out of scope, since we aren't advertising it and the UI ticket is coming up

Okay, where is the UI ticket? I want to copy the comments there. Thanks!

**Review by pllim — APPROVED** [review]
Kyle said he added my comments to the JIRA ticket. So LGTM now. Thanks!

**Pull request merged**